### ISSUE TITLE
How to make `diff` options works on verbose output?

### ISSUE THREAD

**Issue by twpayne** [twpayne/chezmoi#1995]
### Discussed in https://github.com/twpayne/chezmoi/discussions/1994

<div type='discussions-op-text'>

<sup>Originally posted by **latipun7** April  6, 2022</sup>
I set custom diff command in my `.chezmoi.yaml.tmpl`:
```yaml
diff:
  pager: delta
  command: delta
```

It works when the diff is invoked thru `chezmoi diff` or when there's conflict and select `diff` option.

But, I'd like to see custom diff in verbose output too. For example when I run `chezmoi apply -v`, the output currently use built-in git diff format like this:
```shell
❯ chezmoi apply -v ~/.config/git/config
diff --git a/.config/git/config b/.config/git/config
index 7e2fdf2cae5e269c70f781e2b7a157a5133e9117..16319679de505c9e0b8c4c32a6d35b3773312c2a 100600
--- a/.config/git/config
+++ b/.config/git/config
@@ -35,8 +35,6 @@   features = side-by-side line-numbers decorations navigate hyperlinks
   syntax-theme = catppuccin
-  plus-style = syntax "#003800"
-  minus-style = syntax "#3f0001"
 
 [delta "decorations"]
   file-style = bold yellow ul
```
I would like to see the diff with custom diff and the diff pager disabled in verbose output, how can I do that?
If the current state is not available, please make it new features.
Thank you.</div>

**Comment by twpayne** [comment]
@latipun7 I think this change is correct, but the change could do with more testing. Would you be able to check that it invokes `delta` to show the diffs when you run `chezmoi apply -n -v` with your configuration?

**Comment by latipun7** [comment]
Thank you @twpayne ~ This is works, but there's something wrong I guess. To command is exited with exit status 1.
[![asciicast](https://asciinema.org/a/484950.svg)](https://asciinema.org/a/484950)

**Comment by twpayne** [comment]
Thanks for testing and reporting the problem. It turns out that `diff` commands traditionally exit with code 1 if the files differ, and chezmoi was incorrectly propagating this error rather than swallowing it as an expected outcome. I've updated #1996 to fix this.

**Comment by twpayne** [comment]
By the way, re:

```yaml
diff:
  pager: delta
  command: delta
```

you should set either `diff.command` or `diff.pager`, but not both. `delta` expects to receive standard `git diff` output when used as a pager, or generate its own diff output when invoked as a command with arguments, but not both.

**Comment by latipun7** [comment]
> you should set either `diff.command` or `diff.pager`, but not both. `delta` expects to receive standard `git diff` output when used as a pager, or generate its own diff output when invoked as a command with arguments, but not both.

Thanks @twpayne , you are right. But I still want to set both, with the additional args, I now I get what I want, to have `delta` in all commands (`chezmoi diff`, `chezmoi apply` diff option, and `chezmoi apply` verbose output) 👍:
[![asciicast](https://asciinema.org/a/485303.svg)](https://asciinema.org/a/485303)

Anyway, here's my testing result, with the v2.15 behavior as shown above, this is now seeming good in the dev version 👍:
[![asciicast](https://asciinema.org/a/485304.svg)](https://asciinema.org/a/485304)
`chezmoi diff` now showing all diffs, not just exit when found first diff 👍

**Comment by latipun7** [comment]
Sorry to bother again @twpayne . I copied `chezmoi` bin from #1996 to my `/usr/bin/chezmoi`.

Before I tested it to run `chezmoi apply -vn` and didn't try to really apply it (`chezmoi apply -v`). When I try to apply it, the behavior is like a dry run, it didn't change the target. The non-existent file wasn't created, the already existing files weren't overwritten.
[![asciicast](https://asciinema.org/a/485603.svg)](https://asciinema.org/a/485603)
```shell
❯ chezmoi --version
chezmoi version dev
```
